src/runtime_src/xdp/profile/plugin/aie_debug/aie_debug_plugin.cpp

```diff
@@ -0,0 +1,397 @@
+  {
+    std::shared_ptr<xrt_core::device> device = xrt_core::get_userpf_device(handle);
+
+    // Capture all tiles across all graphs
+    // Note: in the future, we could support user-defined tile sets
+    auto graphs = xrt_core::edge::aie::get_graphs(device.get());
+    for (auto& graph : graphs) {
+      mGraphCoreTilesMap[graph] = xrt_core::edge::aie::get_event_tiles(device.get(), graph,
+          xrt_core::edge::aie::module_type::core);
+    }
+
+    // Report tiles (debug only)
+    {
+      std::stringstream msg;
+      msg << "Tiles used for AIE debug:\n";
+      for (const auto& kv : mGraphCoreTilesMap) {
+        msg << kv.first << " : ";
+        for (const auto& tile : kv.second)
+          msg << "(" << tile.col << "," << tile.row << "), ";
+        msg << "\n";
+      }
+      xrt_core::message::send(severity_level::debug, "XRT", msg.str());
+    }
+  }
+
+  std::string AIEDebugPlugin::getCoreStatusString(uint32_t status) {
+    std::string statusStr;
+
+    if (status & 0x1)
+      statusStr += "Enable,";
+    if (status & 0x2)
+      statusStr += "Reset,";
+    if (status & 0x4)
+      statusStr += "Memory_Stall_S,";
+    if (status & 0x8)
+      statusStr += "Memory_Stall_W,";
+    if (status & 0x10)
+      statusStr += "Memory_Stall_N,";
+    if (status & 0x20)
+      statusStr += "Memory_Stall_E,";
+    if (status & 0x40)
+      statusStr += "Lock_Stall_S,";
+    if (status & 0x80)
+      statusStr += "Lock_Stall_W,";
+    if (status & 0x100)
+      statusStr += "Lock_Stall_N,";
+    if (status & 0x200)
+      statusStr += "Lock_Stall_E,";
+    if (status & 0x400)
+      statusStr += "Stream_Stall_SS0,";
+    if (status & 0x800)
+      statusStr += "Stream_Stall_SS1,";
+    if (status & 0x1000)
+      statusStr += "Stream_Stall_MS0,";
+    if (status & 0x2000)
+      statusStr += "Stream_Stall_MS1,";
+    if (status & 0x4000)
+      statusStr += "Cascade_Stall_SCD,";
+    if (status & 0x8000)
+      statusStr += "Cascade_Stall_MCD,";
+    if (status & 0x10000)
+      statusStr += "Debug_Halt,";
+    if (status & 0x20000)
+      statusStr += "ECC_Error_Stall,";
+    if (status & 0x40000)
+      statusStr += "ECC_Scrubbing_Stall,";
+
+    if (!statusStr.empty())
+      statusStr.pop_back();
+
+    return statusStr;
+  }
+
+  void AIEDebugPlugin::pollAIERegisters(uint32_t index, void* handle)
+  {
+    auto it = mThreadCtrlMap.find(handle);
+    if (it == mThreadCtrlMap.end())
+      return;
+
+    // This mask check for following states
+    // ECC_Scrubbing_Stall
+    // ECC_Error_Stall
+    // Debug_Halt
+    // Cascade_Stall_MCD
+    // Cascade_Stall_SCD
+    // Stream_Stall_MS1
+    // Stream_Stall_MS0
+    // Stream_Stall_SS1
+    // Stream_Stall_SS0
+    // Lock_Stall_E
+    // Lock_Stall_N
+    // Lock_Stall_W
+    // Lock_Stall_S
+    // Memory_Stall_E
+    // Memory_Stall_N
+    // Memory_Stall_W
+    // Memory_Stall_S
+    const uint32_t CORE_STALL_MASK = 0xFFFC;
```

Comment:
If all these values are really `const` there should probably be `constexpr`.
At least I appreciate they are not macros already at the first place. ;-)